# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/driver_utils.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/renderer/driver_utils.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/driver_utils.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/driver_utils.h	Sat Sep 07 13:38:36 2019 +0000
@@ -19,6 +19,7 @@
 {
     VENDOR_ID_UNKNOWN = 0x0,
     VENDOR_ID_AMD     = 0x1002,
+    VENDOR_ID_ARM     = 0x13B5,
     VENDOR_ID_INTEL   = 0x8086,
     VENDOR_ID_NVIDIA  = 0x10DE,
     // This is Qualcomm PCI Vendor ID.
@@ -39,6 +40,11 @@
     return vendorId == VENDOR_ID_AMD;
 }
 
+inline bool IsARM(uint32_t vendorId)
+{
+    return vendorId == VENDOR_ID_ARM;
+}
+
 inline bool IsIntel(uint32_t vendorId)
 {
     return vendorId == VENDOR_ID_INTEL;
@@ -123,6 +129,22 @@
 #endif
 }
 
+struct OSVersion
+{
+    OSVersion();
+    OSVersion(int major, int minor, int patch);
+
+    int majorVersion = 0;
+    int minorVersion = 0;
+    int patchVersion = 0;
+};
+bool operator==(const OSVersion &a, const OSVersion &b);
+bool operator!=(const OSVersion &a, const OSVersion &b);
+bool operator<(const OSVersion &a, const OSVersion &b);
+bool operator>=(const OSVersion &a, const OSVersion &b);
+
+OSVersion GetMacOSVersion();
+
 inline bool IsAndroid()
 {
 #if defined(ANGLE_PLATFORM_ANDROID)
@@ -132,5 +154,7 @@
 #endif
 }
 
+int GetAndroidSDKVersion();
+
 }  // namespace rx
 #endif  // LIBANGLE_RENDERER_DRIVER_UTILS_H_